google/resource_google_organization_policy_test.go

```diff
@@ -0,0 +1,311 @@
+package google
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"google.golang.org/api/cloudresourcemanager/v1"
+	"os"
+	"reflect"
+	"testing"
+)
+
+var SOME_ORG_POLICIES = []string{"compute.googleapis.com", "cloudresourcemanager.googleapis.com"}
+
+func TestAccGoogleOrganizationPolicy_boolean_enforced(t *testing.T) {
+	skipIfEnvNotSet(t, "GOOGLE_ORG")
+	org := os.Getenv("GOOGLE_ORG")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleOrganizationPolicyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGoogleOrganizationPolicy_boolean(org, true),
+				Check:  testAccCheckGoogleOrganizationBooleanPolicy("bool", true),
+			},
+		},
+	})
+
+}
+
+func TestAccGoogleOrganizationPolicy_boolean_notEnforced(t *testing.T) {
+	skipIfEnvNotSet(t, "GOOGLE_ORG")
+	org := os.Getenv("GOOGLE_ORG")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleOrganizationPolicyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGoogleOrganizationPolicy_boolean(org, false),
+				Check:  testAccCheckGoogleOrganizationBooleanPolicy("bool", false),
+			},
+		},
+	})
+}
+
+func TestAccGoogleOrganizationPolicy_boolean_update(t *testing.T) {
+	skipIfEnvNotSet(t, "GOOGLE_ORG")
+	org := os.Getenv("GOOGLE_ORG")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleOrganizationPolicyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGoogleOrganizationPolicy_boolean(org, true),
+				Check:  testAccCheckGoogleOrganizationBooleanPolicy("bool", true),
+			},
+			{
+				Config: testAccGoogleOrganizationPolicy_boolean(org, false),
+				Check:  testAccCheckGoogleOrganizationBooleanPolicy("bool", false),
+			},
+			{
+				Config: testAccGoogleOrganizationPolicy_boolean(org, true),
+				Check:  testAccCheckGoogleOrganizationBooleanPolicy("bool", true),
+			},
+		},
+	})
+}
+
+func TestAccGoogleOrganizationPolicy_list_allowAll(t *testing.T) {
+	skipIfEnvNotSet(t, "GOOGLE_ORG")
+	org := os.Getenv("GOOGLE_ORG")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleOrganizationPolicyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGoogleOrganizationPolicy_listAll(org, "allow"),
+				Check:  testAccCheckGoogleOrganizationListPolicyAll("listAll", "ALLOW"),
+			},
+		},
+	})
+}
+
+func TestAccGoogleOrganizationPolicy_list_allowSome(t *testing.T) {
```

Comment:
this test and the deny tests could have side effects with other tests running in parallel

Reply:
I have been able to do 2). However, I haven't found a way to test the denyAll without side-effects but knowing the implementation, if the allowAll works, it is very likely that the denyAll works also, it is not ideal but good enough.

Also, the same methods (`expandListOrganizationPolicy` and `flattenListOrganizationPolicy`) will be used for org policies on folders. Since I can create a new folder easily, I will be able to test the denyAll case without side effects on other tests running in parallel.